### PR TITLE
frontend/feature/logs-events

### PR DESCRIPTION
**Pull request by SanskarGubreley**
## Type of Change

- [ ] Bugfix
- [ x] New feature
- [ x] Enhancement
- [ x] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Frontend/android-native/app/src/main/assets/juspay/sdk_config.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    "logsUrlKey": "https://logs.juspay.in/logs",
+    "logsUrlKeySandbox": "https://beta.logs.juspay.in/logs",
+    "pusherTimerKey": "10.0",
+    "encryptionLevelKey": "encryption",
+    "shouldPush": true,
+    "publicKeySandbox": {
+      "kty": "RSA",
+      "kid": "sandbox-log-24-03-2021",
+      "e": "AQAB",
+      "n": "v9SaTMmbyyPiFEv8NbqQo-y7nmcKLneS8jPkL9Xby04B9xI3wXkezAEjE9dNCd_vs8QEYVJTDIxOsxPTBHc5sm9WzswZMrs7qkr4FAdVpO-UOiiOTBbv8r2GdIR5Do8BFsAAeil1XsiXz15_r2VzawZujjnj6poUbpvfZ3wI2QV4ArVmpoSkGPBKEnpKeILXoz4xZrPCwjaLrAy8qeIWwbm13f0LxSJNHSpk3E0zdf4ZtgzWClqgnJEc4Rq53XNB6h3CaFNBKti5Dcoe3eUABafniOJC4K3gwWwQJ3kQUitpgMy0MnsfGjt2UgqLJTAx8ZZJVrNyFnjYq0QsZibsFw"
```
Comment (witcher-shailesh) [inline Frontend/android-native/app/src/main/assets/juspay/sdk_config.json:12]: Is it authorized or mandatory to put publicKey in repo? 

@kavya-shree-s @Vignesh-772

File: Frontend/ui-common/src/Engineering/Helpers/LogEvent.purs [hunk 2]
```diff
@@ -0,0 +1,85 @@
+module Engineering.Helpers.LogEvent
+  where
```
Comment (witcher-shailesh) [inline Frontend/ui-common/src/Engineering/Helpers/LogEvent.purs:2]: correct the indentation here

File: Frontend/ui-common/src/Engineering/Helpers/LogEvent.purs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import Control.Monad.Except (runExcept)
+import Foreign.Class (class Encode, encode)
+import Foreign.Object (Object, empty, insert, lookup)
+import Tracker (trackActionObject, trackScreenEnd, trackScreen, trackScreenEvent)
+import Tracker.Types (Action(..), Level(..), Screen(..)) as Tracker
+import Prelude (Unit, pure, unit, ($), (<$>), (<<<), (/=), (==), (&&), bind, void, when, discard,map)
+import Presto.Core.Types.Language.Flow (Flow , doAff)
+import JBridge (firebaseLogEvent, firebaseLogEventWithParams, firebaseLogEventWithTwoParams)
+import Log (rootLevelKeyWithRefId)
+import Debug (spy)
```
Comment (witcher-shailesh) [inline Frontend/ui-common/src/Engineering/Helpers/LogEvent.purs:19]: remove debug usage from the code

File: Frontend/ui-customer/src/Flow.purs [hunk 4]
```diff
@@ -81,10 +81,26 @@ import Services.API (AuthType(..), AddressGeometry(..), BookingLocationAPIEntity
 import Services.Backend as Remote
 import Storage (KeyStore(..), deleteValueFromLocalStore, getValueToLocalNativeStore, getValueToLocalStore, isLocalStageOn, setValueToLocalNativeStore, setValueToLocalStore, updateLocalStage)
 import Types.App (ABOUT_US_SCREEN_OUTPUT(..), ACCOUNT_SET_UP_SCREEN_OUTPUT(..), ADD_NEW_ADDRESS_SCREEN_OUTPUT(..), GlobalState(..), CONTACT_US_SCREEN_OUTPUT(..), FlowBT, HELP_AND_SUPPORT_SCREEN_OUTPUT(..), HOME_SCREEN_OUTPUT(..), MY_PROFILE_SCREEN_OUTPUT(..), MY_RIDES_SCREEN_OUTPUT(..), PERMISSION_SCREEN_OUTPUT(..), REFERRAL_SCREEN_OUPUT(..), SAVED_LOCATION_SCREEN_OUTPUT(..), SELECT_LANGUAGE_SCREEN_OUTPUT(..), ScreenType(..), TRIP_DETAILS_SCREEN_OUTPUT(..), EMERGECY_CONTACTS_SCREEN_OUTPUT(..))
+import MerchantConfig.Utils (getMerchant, Merchant(..))
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Flow.purs:84]: remove redundant imports

File: Frontend/ui-customer/src/Screens/OnBoardingFlow/EnterMobileNumberScreen/ScreenData.purs [hunk 5]
```diff
@@ -28,7 +29,8 @@ initData = {
     , otp : ""
     , timer : ""
     , timerID : ""
-    , config : DC.config
+    , config : DC.config,
+    logField : empty
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Screens/OnBoardingFlow/EnterMobileNumberScreen/ScreenData.purs:33]: wrong format

**Comment by SanskarGubreley** [comment]
> Remove the use of `unsafePerformEffect` function as it comes with it's own disadvantages, format the code properly and remove spy statements(if added) or any redundant imports

without using unsafePerformEffect we are not getting logs in firebase, that's why @Vignesh-772 suggested it.

**Pull request closed without merging**